prs/.gitignore

```diff
@@ -0,0 +1,267 @@
+## Ignore Visual Studio temporary files, build results, and
```

Comment:
This PR is pretty large.  I would recommend you split this PR out between feature areas.  Merging in changes to the front end react app, the haskell app and azure functions seems like too much to bite off in one piece.